lightning/src/ln/channelmanager.rs

```diff
@@ -10367,14 +10375,25 @@ where
 				};
 
 				match result {
-					Ok(()) => ResponseInstruction::NoResponse,
-					Err(e) => match responder {
-						Some(responder) => responder.respond(OffersMessage::InvoiceError(e)),
-						None => {
-							log_trace!(self.logger, "No reply path for sending invoice error: {:?}", e);
-							ResponseInstruction::NoResponse
-						},
-					},
+					Err(err) => {
+						match responder {
+							Some(responder) => {
+								abandon_if_payment(context);
+								responder.respond(OffersMessage::InvoiceError(err))
+							}
+							None => {
+								abandon_if_payment(context);
+								log_trace!(
+									self.logger,
+									"A error response was generated, but there is no reply_path specified \
+									for sending the response. Error: {}",
+									err
+								);
+								return ResponseInstruction::NoResponse;
+							}
+						}
+					}
+					Ok(_) => return ResponseInstruction::NoResponse,
```

Comment:
No need for explicit returns here when the expression evaluates to the return value.